planemo/ci.py

```diff
@@ -29,21 +32,16 @@ def filter_paths(ctx, raw_paths, path_type="repo", **kwds):
             diff_dirs = set(os.path.dirname(p) for p in diff_files)
             diff_paths = set()
             for diff_dir in diff_dirs:
-                while diff_dir:
-                    if os.path.isfile(os.path.join(diff_dir, SHED_CONFIG_NAME)):
-                        diff_paths.add(diff_dir)
-                        break
-                    diff_dir = os.path.dirname(diff_dir)
+                diff_path = metadata_file_in_path(diff_dir)
+                if diff_path:
+                    diff_paths.add(diff_path)
+                    break
         else:
             diff_paths = diff_files
 
     unique_paths = set(os.path.relpath(p, cwd) for p in raw_paths)
     if diff_paths is not None:
-        new_unique_paths = []
```

Comment:
I don't think I have understood the purpose here. Was this meant to filter our paths in `diff_paths` that are not within `raw_paths` ?

Reply:
This was (and should remain) the intersection of `unique_paths` (i.e. the tool or repo specified by path by the user) and `diff_paths` (the list of files or directories changed in Git).
Sorry I missed this at the time, should be fixed by https://github.com/galaxyproject/planemo/pull/1127 .